great_expectations/datasource/fluent/fluent_base_model.py

```diff
@@ -321,7 +576,9 @@ def _recursively_set_config_value(  # noqa: C901 - too complex
                 _recursively_set_config_value(v, config_provider)
 
 
-def _update__fields_set__on_truthyness(model: FluentBaseModel, field_name: str) -> None:
```

Comment:
Benefit of the above suggestion: We don't need union types here.

Reply:
[ef4651e](https://github.com/great-expectations/great_expectations/pull/9785/commits/ef4651e80793253928d7cf9e8755cd9b2c74debc)